File: Encryption.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the 'encryption' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

string encryption(string s)
{
    std::string result{};
    s.erase(std::remove(s.begin(), s.end(), ' '), s.end());
    size_t L = s.size();

    size_t col = (size_t)std::ceil(std::sqrt(L));

    for (size_t i = 0; i < col; ++i)
    {
        for (size_t j = i; j < L; j += col)
        {
            result.append(1, s[j]);
        }

        result.append(1, ' ');
    }

    return result;
}
int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = encryption(s);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: Encryption.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'encryption' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts STRING s as parameter.
     */

    public static string encryption(string s)
    {
        s = s.Replace(" ", "");

        int L = s.Length;

        int col = (int)Math.Ceiling(Math.Sqrt(L));

        StringBuilder result = new StringBuilder();

        for (int i = 0; i < col; ++i)
        {
            for (int j = i; j < L; j += col)
            {
                result.Append(s[j]);
            }

            result.Append(' ');
        }

        return result.ToString();
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        string s = Console.ReadLine();

        string result = Result.encryption(s);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}
